src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -8380,6 +8380,48 @@ internal static string HandleDoubleAndSingleQuote(ref string wordToComplete)
             return quote;
         }
 
+        /// <summary>
+        /// Enumerates quoted and unquoted completion text.
+        /// This makes it easier to handle different variations of completions with consideration of quotes.
+        /// </summary>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="possibleCompletionValues">The possible completion values to iterate.</param>
+        /// <param name="toolTipMapping">The optional tool tip mapping delegate.</param>
+        /// <param name="resultType">The optional completion result type. Default is Text.</param>
+        /// <returns></returns>
+        internal static IEnumerable<CompletionResult> EnumerateQuotedAndUnquotedCompletionText(
+            string wordToComplete,
+            IEnumerable<string> possibleCompletionValues,
+            Func<string, string> toolTipMapping = null,
+            CompletionResultType resultType = CompletionResultType.Text)
+        {
+            string quote = HandleDoubleAndSingleQuote(ref wordToComplete);
+            var pattern = WildcardPattern.Get(wordToComplete + "*", WildcardOptions.IgnoreCase);
+
+            foreach (string value in possibleCompletionValues)
+            {
+                if (pattern.IsMatch(value))
+                {
+                    string completionText = quote == string.Empty
+                        ? value
+                        : quote + value + quote;
+
+                    if (toolTipMapping is not null)
+                    {
+                        yield return new CompletionResult(
+                            completionText,
+                            listItemText: value,
+                            resultType,
+                            toolTip: toolTipMapping(value));
+                    }
+                    else
+                    {
+                        yield return new CompletionResult(completionText);
```

Comment:
```suggestion
                        yield return new CompletionResult(completionText, value, resultType, value);
```
This is so the specified resultType is actually set.

Reply:
Thanks. I've just removed the if/else and just toggled the tool tip. Fixed in b0a9dff8914cb2421f39a1366eea2eaae052fd78.